modules/signals/src/state-source.ts

```diff
@@ -49,6 +54,10 @@ export function patchState<State extends object>(
   notifyWatchers(stateSource);
 }
 
+function freezeInDevMode<State extends object>(value: State): State {
+  return ngDevMode ? deepFreeze(value) : value;
+}
+
```

Comment:
`freezeInDevMode` (and `ngDevMode`) can be moved to the `deep-freeze.ts` file as well. Since it's not directly related to the state object, the generic can be named `T`:

```suggestion
function freezeInDevMode<T>(target: T): T {
  return ngDevMode ? deepFreeze(target) : target;
}
```